### PR TITLE
User sign up

### LINKED ISSUE #825: Signup UI
Most of it is already there I think, but it needs to be finished and hooked up.

### PR DESCRIPTION
**Pull request by willsb**
Closes #825 

Squash whenever

### CODE REVIEW

File: Toggl.Foundation/Login/LoginManager.cs [hunk 1]
```diff
@@ -51,6 +51,25 @@ public IObservable<ITogglDataSource> Login(Email email, string password)
                     });
         }
 
+        public IObservable<ITogglDataSource> SignUp(Email email, string password)
+        {
+            if (!email.IsValid)
+                throw new ArgumentException("A valid email must be provided when trying to Login");
```
Comment (amulware) [inline Toggl.Foundation/Login/LoginManager.cs]: 'signup', not 'login'

File: Toggl.Ultrawave/ApiClients/UserApi.cs [hunk 2]
```diff
@@ -54,6 +54,7 @@ public IObservable<IUser> SignUp(Email email, string password)
             return CreateObservable<User>(endPoints.Post, new HttpHeader[0], json);
         }
 
+        [Preserve(AllMembers = true)]
```
Comment (amulware) [inline Toggl.Ultrawave/ApiClients/UserApi.cs]: Why is this necessary? The classes are used just above.

Also, is there a reason that we are not simply skipping this whole project for link-stripping purposes?

Comment (willsb) [reply Toggl.Ultrawave/ApiClients/UserApi.cs]: The default constructor was being stripped away, dunno why

We have discussed this skipping process and opted for make the libraries linker safe instead. This approach is better in the long run

Comment (amulware) [reply Toggl.Ultrawave/ApiClients/UserApi.cs]: Noted.
